src/main/java/fr/cnes/sonar/plugins/icode/check/ICodeSensor.java

```diff
@@ -150,12 +151,25 @@ protected void executeICode(final SensorContext sensorContext) {
         LOGGER.info("i-Code CNES auto-launch enabled.");
         final Configuration config = sensorContext.config();
         final String executable = config.get(ICodePluginProperties.ICODE_PATH_KEY).orElse(ICodePluginProperties.ICODE_PATH_DEFAULT);
-        final String[] files = sensorContext.fileSystem().baseDir().list();
+
+        ArrayList<String> filesToUse = new ArrayList<String>();
+
+	FilePredicates p = sensorContext.fileSystem().predicates();
+
+	final Iterable<InputFile> files = sensorContext.fileSystem().inputFiles(p.or(p.hasLanguage("shell"),p.hasLanguage("f70"),p.hasLanguage("f90")));
+	//sensorContext.fileSystem().predicates().all());
```

Comment:
No commented code should be present (we already use Git to save old code version).